src/main/fc/fc_rc.c

```diff
@@ -298,24 +307,36 @@ FAST_CODE uint8_t processRcSmoothingFilter(void)
             if (rxIsReceivingSignal() && (targetPidLooptime > 0) && (millis() > RC_SMOOTHING_FILTER_TRAINING_DELAY_MS)) {
                 rxFrameTimeSum += currentRxRefreshRate;
                 rxFrameCount++;
+                maxRxFrameInterval = MAX(maxRxFrameInterval, currentRxRefreshRate);
+                minRxFrameInterval = MIN(minRxFrameInterval, currentRxRefreshRate);
+                DEBUG_SET(DEBUG_RC_SMOOTHING, 0, rxFrameCount);           // log the step count during training
+                DEBUG_SET(DEBUG_RC_SMOOTHING, 3, currentRxRefreshRate);   // log each frame interval during training
                 if (rxFrameCount >= RC_SMOOTHING_FILTER_TRAINING_SAMPLES) {
-                    const float avgRxFrameRate = rxFrameTimeSum / rxFrameCount / 1000;
-                    defaultInputCutoffFrequency = calcRcSmoothingCutoff(avgRxFrameRate, RC_SMOOTHING_FILTER_INPUT_AUTO_HZ);
-                    defaultDerivativeCutoffFrequency = calcRcSmoothingCutoff(avgRxFrameRate, RC_SMOOTHING_FILTER_DERIVATIVE_AUTO_HZ);
+                    rxFrameTimeSum = rxFrameTimeSum - minRxFrameInterval - maxRxFrameInterval; // Throw out high and low samples
+                    calculatedFrameAverageMs = lrintf(rxFrameTimeSum / (rxFrameCount - 2));
+                    const float avgRxFrameRate = (rxFrameTimeSum / (rxFrameCount - 2)) / 1000;
```

Comment:
It's semantically incorrect and misleading to call this variable **rate**, by convention rate has dimension of inverse time, while this one is `time`. Let's call it `avgRxFrameTime`.
The variable above on the other hand has no signs of either *rate* or *time* at all, `calculatedFrameTimeAverageMs`?